src/parser/Lexer.mjs

```diff
@@ -887,7 +888,7 @@ export class Lexer {
       }
       const c = this.source[this.position];
       if (isRegularExpressionFlagPart(c)
-          && 'gimsuy'.includes(c)
+          && (surroundingAgent.feature('regexp-match-indices') ? 'dgimsuy' : 'gimsuy').includes(c)
```

Comment:
you can use `this.feature` instead.

Reply:
If only there were a way to indicate `Lexer` had an abstract method named `feature` that must eventually be implemented by a subclass like `Parser`... 😉